### DISCUSSION TITLE
Version 0.76 -> 0.85 Snowflake Schema Object Wants To Be Recreated

### DISCUSSION THREAD

**Discussion by mcurtisslalom** [Snowflake-Labs/terraform-provider-snowflake#3270] [Q&A]
Has anyone run into an issue where it schemas want to be recreated between these versions, despite being present in the state list? The snowflake schema object is contained within a module. The module creates a schema along with provisioned access roles for later use by functional roles.

This doesn't strike me as a bug but, more likely me not following the correct procedure to update. Thus, I have posted it as a discussion.

terraform apply:
```
  # module.tf_<db_name>_<schema_name>_snowflake_schema.schema will be created
  + resource "snowflake_schema" "schema" {
      + data_retention_days = 1
      + database            = "<db_name>"
      + id                  = (known after apply)
      + is_managed          = false
      + is_transient        = false
      + name                = "<schema_name"
    }
```
    
Import results in error:
  Error: Resource already managed by Terraform

**Comment by sfc-gh-jcieslak** [comment]
Hey @mcurtisslalom 👋 
Please take a look at this thread: https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/3015, and let me know if the provided fix helps with your case. tldr; Because of the migration limitations `is_transient` has to be set explicitly if it wasn't before. It may be wrong in this instance, if so, I would need more logs because that's not enough. Let's try the fix from the provided issue and If it won't work we'll dive deeper.

**Reply by mcurtisslalom** [reply]
Hey I actually just figured it out this morning. It was a permission issue on the database. I updated our database module to use "snowflake_grant_privileges_to_account_role" when handing out access permissions within the module. I used "all_privileges = true"  which only gave out "APPLYBUDGET, CREATE DATABASE ROLE, MODIFY". Losing "USAGE" and "CREATE SCHEMA". When I explicitly listed permissions instead of using the flag the access role was given the necessary permissions. Terraform "wanting to recreate schemas" was under the hood "the provider role doesn't have permissions to see the schema".

Thanks for the the follow up though! I will be sure to update the is transient flag before reworking the schemas module :)

**Reply by sfc-gh-asawicki** [reply]
Hey @mcurtisslalom. Do you have any more questions or is there any follow-up needed here?
